tracer/test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -420,32 +423,21 @@ private void HandlePotentialTraces(MockHttpParser.MockHttpRequest request)
                 {
                     var i = 0;
                     body = new byte[request.ContentLength];
-                    var moreBytesThanContentLength = false;
-                    while (request.Body.Stream.CanRead)
+
+                    while (request.Body.Stream.CanRead && i < request.ContentLength)
                     {
                         var nextByte = request.Body.Stream.ReadByte();
 
                         if (nextByte == -1)
                         {
                             break;
                         }
-                        else if (i < request.ContentLength)
-                        {
-                            body[i] = (byte)nextByte;
-                        }
-                        else
-                        {
-                            moreBytesThanContentLength = true;
-                        }
 
+                        body[i] = (byte)nextByte;
                         i++;
                     }
 
-                    if (moreBytesThanContentLength)
-                    {
-                        throw new Exception($"More bytes were sent than we counted. {i} read versus {request.ContentLength} expected.");
```

Comment:
We should probably still care about this shouldn't we, as it would flag an error in our agentwriter code?

Reply:
Yeah but I don't know how to do both. If we keep reading and the client sends nothing else and is expecting the response (which is the expected behavior), well we'll keep waiting forever 😛  that's why I had to change this

Reply:
Oh I can use the extra receive call for that 👍

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2337/commits/367324bbb235e0a48e538b54b6e16f8c944a95b6